pkg/codegen/schema/schema.go

```diff
@@ -1515,15 +1551,57 @@ type ResourceSpec struct {
 	Methods map[string]string `json:"methods,omitempty" yaml:"methods,omitempty"`
 }
 
+// ReturnTypeSpec is either ObjectTypeSpec or TypeSpec
+type ReturnTypeSpec struct {
+	ObjectTypeSpec *ObjectTypeSpec
+	TypeSpec       *TypeSpec
+}
+
+func (returnTypeSpec *ReturnTypeSpec) UnmarshalJSON(inputJSON []byte) error {
```

Comment:
Unfortunately, we also need an `UnmarshalYAML`.

Reply:
added the implementation, similar to UnmarshalJSON